csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetPackages.cs

```diff
@@ -47,6 +50,41 @@ public NugetPackages(string sourceDir, TemporaryDirectory packageDirectory, Util
             {
                 logger.LogInfo($"Found {packageFiles.Length} packages.config files, trying to use nuget.exe for package restore");
                 nugetExe = ResolveNugetExe(sourceDir);
+                if (HasNoPackageSource())
+                {
+                    // We only modify or add a top level nuget.config file
+                    nugetConfigPath = Path.Combine(sourceDir, "nuget.config");
```

Comment:
We should preserve the casing of the nuget.config file in the repo, otherwise when we move the backup to `nugetConfigPath`, we might change the casing, which would mean a modification to the working directory.